internal/plugin/grpc_client_plugin.go

```diff
@@ -10,64 +10,79 @@ import (
 	"fmt"
 	"log/slog"
 	"net"
-	"time"
+	"sync"
 
 	"github.com/google/uuid"
-	grpcRetry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
 	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/internal/backoff"
 	"github.com/nginx/agent/v3/internal/bus"
 	"github.com/nginx/agent/v3/internal/config"
+	agentGrpc "github.com/nginx/agent/v3/internal/grpc"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials/insecure"
-	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/connectivity"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	uuidLibrary "github.com/nginx/agent/v3/internal/uuid"
 )
 
-const (
-	DefaultClientTime   = 120  // Default time for client operations in seconds
-	DefaultTimeout      = 60   // Default timeout duration in seconds
-	DefaultPermitStream = true // Flag indicating permission of stream
-)
-
 type (
 	GrpcClient struct {
-		messagePipe bus.MessagePipeInterface
-		config      *config.Config
-		conn        *grpc.ClientConn
-		cancel      context.CancelFunc
+		messagePipe    bus.MessagePipeInterface
+		config         *config.Config
+		conn           *grpc.ClientConn
+		cancel         context.CancelFunc
+		connectionLock sync.Mutex
```

Comment:
Very minor but should `connectionLock` be renamed to `connectionMutex` to be consistent with the naming used for other mutex in the codebase?